File: deck-o-cards.js

function deck_o_cards() {
  var values = [2, 3, 4, 5, 6, 7, 8, 9, 10, 'Jack', 'Queen', 'King', 'Ace'];
  var suits = ['hearts', 'diamonds', 'clubs', 'spades'];

  var cards = []; // deck
  var shuffledCards = []; // deck shuffled
  var pokerHand = []; // poker hand
  var objIndex = 0; //card counter
  var pokerStatement = "Your poker hand is a(n)"; //string to tell you your poker hand

  // Makes 52 card objects and store them in the "cards" array
  for (i=0; i < suits.length; i++){
    for (p=0; p < values.length; p++){
      cards[objIndex] = {suit: suits[i], value: values[p]};
      objIndex += 1;
    };
  };

  // Shuffles the cards
  function shuffle(array) {
      var counter = array.length, temp, index;
      // While there are elements in the array
      while (counter > 0) {
          // Pick a random index
          index = Math.floor(Math.random() * counter);
          // Decrease counter by 1
          counter--;
          // And swap the last element with it
          temp = array[counter];
          array[counter] = array[index];
          array[index] = temp;
      }
      return array;
  };

  shuffledCards = shuffle(cards);

  //Makes poker hand
  for (i=0; i < 5; i++){
    pokerHand[i] = shuffledCards[i];
  };

  for(i=0; i < pokerHand.length; i++){
    if (i == pokerHand.length-1){
      pokerStatement += "and a(n) " + pokerHand[i].value + " of " + pokerHand[i].suit + ". ";
      break;
    }
    pokerStatement += " " + pokerHand[i].value + " of " + pokerHand[i].suit + ",";
  }

  // Pull the top card from the newly shuffledCards

  var topCard = shuffledCards[0];

  // 3. Print the results:
  // "The deck has {n} cards"
  // "The top card is the {value} of {suit}"

  console.log("The deck has " + objIndex + " cards");
  console.log("The top card is the " + topCard.value + " of " + topCard.suit);
  console.log();
  console.log(pokerStatement);

};

File: deck-o-cards.py
from random import random

class DeckOfCards():
    def __init__(self, suits, values):
        self.cards = []
        for i in range(0, len(suits)):
            for j in range(0, len(values)):
                self.cards.append({suits[i], values[j]})
        self.quantity = len(self.cards)
        print("The deck has {} cards.".format(self.quantity))
        self.shuffle(self.cards)
        print("The top card is a {}.".format(self.cards[0]))

    def shuffle(self,list):
        counter = len(list)
        while counter > 0:
            index = int(random()*counter)
            counter-=1
            temp=list[counter]
            list[counter] = list[index]
            list[index]=temp
        return list

    def poker_hand(self):
        print("Your poker hand is: ")
        hand = self.cards[:5]
        for i in range(0,len(hand)):
            print(hand[i])
        return hand



suits = ['hearts', 'diamonds', 'clubs', 'spades']
values = [2, 3, 4, 5, 6, 7, 8, 9, 10, 'Jack', 'Queen', 'King', 'Ace']


my_deck = DeckOfCards(suits, values)
my_hand = my_deck.poker_hand()
